src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -784,6 +790,67 @@ public void deleteRef(String refName) throws GitException, InterruptedException
         return references;
     }
 
+    @Override
+    public Map<String, String> getRemoteSymbolicReferences(String url, String pattern)
+            throws GitException, InterruptedException {
+        Map<String, String> references = new HashMap<>();
+        String regexPattern = null;
+        if (pattern != null) {
+            regexPattern = createRefRegexFromGlob(pattern);
```

Comment:
This call to createRefRegexFromGlob creates a regex pattern which won't match "HEAD".  Line 805 seems to mandate that only HEAD is accepted as a parameter to JGit for non-empty results, but if I pass "HEAD", the results are empty for JGit but have a value for CliGit.

Refer to e89785033c15b1fec3fcb4165757a966baffc74c for a test to show the problem, and to aa92d89e7e4fd7c90c01b7e1f0e6cc210cd72874 for one alternative to fix it.